app/app.go

```diff
@@ -399,7 +399,7 @@ func GetMaccPerms() map[string][]string {
 
 func (app *TerraApp) setupUpgradeHandlers() {
 	app.UpgradeKeeper.SetUpgradeHandler(
-		v2.UpgradeName,
-		v2.CreateV2UpgradeHandler(app.mm, app.configurator),
+		v3.UpgradeName,
```

Comment:
you just need to add new

```
app.UpgradeKeeper.SetUpgradeHandler(
		v3.UpgradeName,
                 v3.CreateV3UpgradeHandler(app.mm, app.configurator),
)
```

and keep v2 intact

Reply:
https://github.com/osmosis-labs/osmosis/blob/main/app/app.go#L447

like here, osmosis use a loop to add all upgrade version handler

Reply:
I like the Osmosis approach

Reply:
Implemented the requested code changes.